oeps/oep-0045-arch-ops-and-config.rst

```diff
@@ -0,0 +1,175 @@
+##########################################
+OEP-45: Configuring and Operating Open edX 
+##########################################
+
+.. list-table::
+
+   * - OEP
+     - :doc:`OEP-0045 </oeps/oep-0045>`
+   * - Title
+     - Configuring and Operating Open edX
+   * - Last Modified
+     - 2020-04-06
+   * - Authors
+     - Bill DeRusha
+   * - Arbiter
+     - Felipe Montoya 
+   * - Status
+     - Under Review
+   * - Type
+     - Architecture
+   * - Created
+     - 2020-04-06
+   * - Review Period
+     - 2020-04-17 - 2020-05-04
+
+.. contents::
+   :local:
+   :depth: 3
+
+
+Abstract
+========
+
+Wherever possible, The edX organization will provide and manage Docker images as the medium for packaging operating dependencies of Open edX applications rather than providing Ansible playbooks.
+
+The configuration for running applications in different environments will be done through a config file per unique [application, environment] combination. How those files are generated and managed is the responsibility of each operator.
+
+All standard operations for running and managing any Open edX application will be documented. How those procedures are executed is the responsibility of each operator.
+
+Wherever possible, operating concerns will live within the codebase they are concerned with. Thus it should be expected that all applications will provide the following:
+
+* A Dockerfile from which the container image can be built.
+* A documented settings file with production-ready defaults.
```

Comment:
What does "production-ready" mean exactly? I assume we don't mean, "appropriate for edx.org production"?

Reply:
openedx production ready.  meaning, we would expect an openedx user to run the image on a public website and things like DDT or verbose logging or dev user accounts wouldn't be present.

Reply:
@nedbat it's a squishy word on purpose.  It means production ready as opposed to development ready.  A great example of how this would look was highlighted by Regis. 

`DEBUG = false` is a production-ready default.  `DEBUG = true` is not.

But to directly answer you, production-ready means reasonable to run on "most/many" production systems.  Anything edx.org specific would not pass this bar.